boards/arm/reel_board/reel_board.dts

```diff
@@ -52,6 +52,33 @@
 		};
 	};
 
+	arduino_header: connector {
+		compatible = "arduino-header-r3";
```

Comment:
I overlooked this part: Are you sure this should be an arduino connector?
Don't you want to use a new specific connector type (edge connector)?

Reply:
It should be arduino connector, there is an adapter `reel board edge connector -> arduino` (link board BASE). The docu from the reel board is a bit behind.